core/src/main/scala/caliban/execution/Fragment.scala

```diff
@@ -0,0 +1,29 @@
+package caliban.execution
+
+import caliban.Value.{ BooleanValue, IntValue, StringValue }
+import caliban.parsing.adt.Directive
+
+case class Fragment(name: Option[String], directives: List[Directive]) {}
+
+object Fragment {
+  object IsDeferred {
+    def unapply(fragment: Fragment): Option[Option[String]] =
+      fragment.directives.collectFirst {
+        case Directive("defer", args, _) if args.get("if").forall {
+              case BooleanValue(v) => v
+              case _               => true
+            } =>
+          args.get("label").collect { case StringValue(v) => v }
+      }
+  }
+}
+
+object IsStream {
+  def unapply(field: Field): Option[(Option[String], Option[Int])] =
+    field.directives.collectFirst { case Directive("stream", args, _) =>
+      (
+        args.get("label").collect { case StringValue(v) => v },
+        args.get("initialCount").collect { case v: IntValue => v.toInt }
+      )
+    }
+}
```

Comment:
Should this be inside the `Fragment` object?

Reply:
No I don't think so. The stream is applied to fields instead of fragments even though it does similar things